### PR TITLE
Default to $home/scooze for storage of bulk data and logs

### LINKED ISSUE #292: BUG: Can't Run scooze Commands Outside Package Root
**Describe the bug**
Unable to run scooze commands like `scooze load cards --oracle` outside of the package root. Doing so will result in an error reading from the logging config.

**To Reproduce**
Run `scooze load cards --oracle` or any other scooze command from any folder except for the root.

**Expected behavior**
These commands should work from any environment that has scooze installed.

**Additional context**
If you delete the `logs/` directory, there is a similar issue with writing to the log file. Is it possible to address this as well?

### PR DESCRIPTION
**Pull request by fjork3**

### CODE REVIEW

File: src/scooze/__init__.py [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
 from pathlib import Path
 
+# Set Up Logging
 logger = logging.getLogger(__name__)
 
-config_file = Path("configs/logging_config.json")
-with open(config_file) as f_in:
+# TODO: move this to utils or the top of this file as an ENV var or something?
+package_root = Path(__file__).parent
+
```
Comment (iambroadband) [inline src/scooze/__init__.py:12]: Is this a flag we want available elsewhere? Can I change this to be ALL_CAPS?

File: src/scooze/console/__init__.py [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+# it's not working as intended right now, so the current behavior will store the log wherever the script is run
+# is this the desired behavior?
+log_dir = Path.home() / "scooze" / "logs"
+log_dir.mkdir(parents=True, exist_ok=True)
+log_path = log_dir / "scooze.log.jsonl"
+
+for handler in logger.handlers:
+    if handler.name == "file":
+        print(f"update {handler.name} to use {log_path}")
+        handler.baseFilename = log_path
```
Comment (iambroadband) [inline src/scooze/console/__init__.py]: This code does nothing and can be removed unless we intend to fix it and make it functional.

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,11 @@
 - Refactored the bulkdata API endpoints ([#279](https://github.com/arcavios/scooze/pull/279))
 - Renamed DeckPart to CardList ([#281](https://github.com/arcavios/scooze/pull/281))
 - Use `pathlib` for consistent cross-OS file paths ([#222](https://github.com/arcavios/scooze/pull/222))
+- Default to home/scooze for storage of bulk data and logs ([#294](https://github.com/arcavios/scooze/pull/294))
+
+
+#### Fixed
+
```
Comment (iambroadband) [inline CHANGELOG.md:22]: Can remove this since we moved it to the Changed section.

**Comment by iambroadband** [comment]
tests are breaking like this:

``` shell
╰─❯ pytest
/home/broadband/.cache/pypoetry/virtualenvs/scooze-iup8854z-py3.11/lib/python3.11/site-packages/_pytest/config/__init__.py:331: PluggyTeardownRaisedWarning: A plugin raised an exception during an old-style hookwrapper teardown.
Plugin: helpconfig, Hook: pytest_cmdline_parse
ConftestImportFailure: ValueError: Unable to configure handler 'file' (from /home/broadband/workspace/arcavios/scooze/tests/conftest.py)
For more information see https://pluggy.readthedocs.io/en/stable/api_reference.html#pluggy.PluggyTeardownRaisedWarning
  config = pluginmanager.hook.pytest_cmdline_parse(
ImportError while loading conftest '/home/broadband/workspace/arcavios/scooze/tests/conftest.py'.
tests/conftest.py:12: in <module>
    from scooze.card import OracleCard
src/scooze/__init__.py:16: in <module>
    logging.config.dictConfig(config=logging_config)
../../../.pyenv/versions/3.11.5/lib/python3.11/logging/config.py:823: in dictConfig
    dictConfigClass(config).configure()
../../../.pyenv/versions/3.11.5/lib/python3.11/logging/config.py:580: in configure
    raise ValueError('Unable to configure handler '
E   ValueError: Unable to configure handler 'file'
```

**Comment by iambroadband** [comment]
> tests are breaking like this:
> 
> ```shell
> ╰─❯ pytest
> /home/broadband/.cache/pypoetry/virtualenvs/scooze-iup8854z-py3.11/lib/python3.11/site-packages/_pytest/config/__init__.py:331: PluggyTeardownRaisedWarning: A plugin raised an exception during an old-style hookwrapper teardown.
> Plugin: helpconfig, Hook: pytest_cmdline_parse
> ConftestImportFailure: ValueError: Unable to configure handler 'file' (from /home/broadband/workspace/arcavios/scooze/tests/conftest.py)
> For more information see https://pluggy.readthedocs.io/en/stable/api_reference.html#pluggy.PluggyTeardownRaisedWarning
>   config = pluginmanager.hook.pytest_cmdline_parse(
> ImportError while loading conftest '/home/broadband/workspace/arcavios/scooze/tests/conftest.py'.
> tests/conftest.py:12: in <module>
>     from scooze.card import OracleCard
> src/scooze/__init__.py:16: in <module>
>     logging.config.dictConfig(config=logging_config)
> ../../../.pyenv/versions/3.11.5/lib/python3.11/logging/config.py:823: in dictConfig
>     dictConfigClass(config).configure()
> ../../../.pyenv/versions/3.11.5/lib/python3.11/logging/config.py:580: in configure
>     raise ValueError('Unable to configure handler '
> E   ValueError: Unable to configure handler 'file'
> ```

fixed

**Comment by iambroadband** [comment]
gonna request from @cmentzer or @JacobGinsparg since me and Ben both have too many commits to review this lol

File: src/scooze/__init__.py [hunk 4]
```diff
@@ -1,13 +1,14 @@
 import json
-import logging
 import logging.config
+import logging.handlers
```
Comment (iambroadband) [inline src/scooze/__init__.py]: I think we still need to import logging.

File: src/scooze/utils.py [hunk 5]
```diff
@@ -57,6 +58,34 @@
...
+        delay: bool = False,
+        errors: str | None = None,
+    ):
+        path = Path.home() / "scooze" / "logs"
+        path.mkdir(parents=True, exist_ok=True)
+        super(ScoozeRotatingFileHandler, self).__init__(
+            filename=path / filename,
+            mode=mode,
+            maxBytes=maxBytes,
+            backupCount=backupCount,
```
Comment (iambroadband) [inline src/scooze/utils.py:82]: Weird whitespace

**Review by iambroadband — DISMISSED** [review]

File: src/scooze/utils.py [hunk 6]
```diff
@@ -57,6 +58,34 @@
...
+        self,
+        filename: str,
+        mode: str = "a",
+        maxBytes: int = 0,
+        backupCount: int = 0,
+        encoding: str | None = None,
+        delay: bool = False,
+        errors: str | None = None,
+    ):
+        path = Path.home() / "scooze" / "logs"
```
Comment (fjork3) [inline src/scooze/utils.py]: Should we be taking this as a parameter? I'd expect some folks would want to use a different log location, and I'm not sure where else they'd be setting it.

Comment (iambroadband) [reply src/scooze/utils.py]: We can, but I don't think we _have_ to. Loggers in applications using scooze can handle their logs however they want.

File: src/scooze/utils.py [hunk 7]
```diff
@@ -13,8 +14,8 @@
 from scooze.catalogs import CostSymbol, Format
 from scooze.enum import ExtendedEnum
 
-DEFAULT_BULK_FILE_DIR = str(Path("./data/bulk"))
-DEFAULT_DECKS_DIR = str(Path("./data/decks"))
+DEFAULT_BULK_FILE_DIR = str(Path.home() / "scooze" / "data" / "bulk")
+DEFAULT_DECKS_DIR = str(Path.home() / "scooze" / "data" / "decks")
```
Comment (iambroadband) [inline src/scooze/utils.py]: lets put this in a hidden folder `.scooze` instead

**Review by JacobGinsparg — APPROVED** [review]

**Pull request merged**